### PR TITLE
Fix unittest/mock.pyi

### PR DESCRIPTION
**Pull request by smessmer**
Add a lot of missing types

### CODE REVIEW

File: stdlib/3/unittest/mock.pyi [hunk 1]
```diff
@@ -10,60 +10,60 @@ if sys.version_info >= (3, 3):
...
-        def __getattr__(self, name): ...
-        def __dir__(self): ...
-        def __setattr__(self, name, value): ...
-        def __delattr__(self, name): ...
-        def assert_not_called(_mock_self): ...
-        def assert_called_with(_mock_self, *args, **kwargs): ...
-        def assert_called_once_with(_mock_self, *args, **kwargs): ...
-        def assert_has_calls(self, calls, any_order=False): ...
-        def assert_any_call(self, *args, **kwargs): ...
+        def reset_mock(self, visited:bool=None) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/3/unittest/mock.pyi]: I think typeshed style puts a space after : and around every =.

File: stdlib/3/unittest/mock.pyi [hunk 2]
```diff
@@ -10,60 +10,60 @@ if sys.version_info >= (3, 3):
...
     class Base:
-        def __init__(self, *args, **kwargs): ...
+        def __init__(self, *args: Any, **kwargs: Any) -> None: ...
 
-    class NonCallableMock(Base):
-        def __new__(cls, *args, **kw): ...
-        def __init__(self, spec=None, wraps=None, name=None, spec_set=None, parent=None, _spec_state=None, _new_name='', _new_parent=None, _spec_as_instance=False, _eat_self=None, unsafe=False, **kwargs): ...
-        def attach_mock(self, mock, attribute): ...
-        def mock_add_spec(self, spec, spec_set=False): ...
+    class NonCallableMock(Any):
```
Comment (JelleZijlstra) [inline stdlib/3/unittest/mock.pyi]: Inheriting from Any is an interesting choice. I'm not sure what type checkers will make of it.

File: stdlib/3/unittest/mock.pyi [hunk 3]
```diff
@@ -78,74 +78,76 @@ if sys.version_info >= (3, 3):
...
-        def __eq__(self, other): ...
-        def __ne__(self, other): ...
+        def __eq__(self, other:Any)->bool: ...
+        def __ne__(self, other:Any)->bool: ...
 
     ANY = ...  # type: Any
 
     class _Call(tuple):
-        def __new__(cls, value=..., name=None, parent=None, two=False, from_kall=True): ...
+        def __new__(cls, value:Any=..., name:Any=None, parent:Any=None, two:Any=False, from_kall:Any=True)->Any: ...
```
Comment (JelleZijlstra) [inline stdlib/3/unittest/mock.pyi]: `two` and `from_kall` can be typed as `bool` I think.

File: stdlib/3/unittest/mock.pyi (hunk 1 above)
Comment (gvanrossum) [inline stdlib/3/unittest/mock.pyi]: Yes, that's in PEP 8 (and the examples in PEP 484 are also formatted that way).

File: stdlib/3/unittest/mock.pyi (hunk 2 above)
Comment (smessmer) [inline stdlib/3/unittest/mock.pyi]: It's the only way I was able to
- Have a separate type `Mock`, i.e. defining `Mock` as type in functions
- Passing an instance of `Mock` to functions expecting other classes (i.e. using the mock for its purpose)

**Pull request merged**